src/app/shared/services/backend/link-preview.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core'
+import {HttpClient, HttpParams} from '@angular/common/http'
+import {Observable} from 'rxjs'
+import {environment} from '../../../../environments/environment'
+import {map} from "rxjs/operators"
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LinkPreviewService {
+  path = `${environment.backendURL}/api/link`
+
+  constructor(private http: HttpClient) {
+  }
+
+  previewLink(url: string): Observable<LinkPreviewResponse> {
+    const params = new HttpParams().set('url', url)
+    return this.http.get<LinkPreviewResponse>(`${this.path}/preview`, {params: params}).pipe(
```

Comment:
You can use BackendHttpClient in this case. If using it, you can use its get method to set parameters as a second parameter so you don't need line 17.